### ISSUE TITLE
Exception in thread "main" com.fasterxml.jackson.databind.JsonMappingException: Document nesting depth (1001) exceeds the maximum allowed 

### ISSUE THREAD

**Issue by Fabi9199** [nvuillam/npm-groovy-lint#406]
Hey

I am facing an issue during linting of several groovy files I am using in a Jenkins Pipeline.
The log containing the exception can be seen here:
[Exception.log](https://github.com/user-attachments/files/16714533/Exception.log)

I am using the groovy linter now in several projects but only face this problem in one of them, it is a Jenkins Library.
What could be the problem causing this?

Thanks and best regards,
Fabian

**Comment by nvuillam** [comment]
@Fabi9199 please can you try with npm-groovy-lint@beta ?
I updated to latest CodeNarc version, that contains Jenkinsfile ruleset, maybe it will work better

**Comment by Fabi9199** [comment]
> @Fabi9199 please can you try with npm-groovy-lint@beta ? I updated to latest CodeNarc version, that contains Jenkinsfile ruleset, maybe it will work better

@nvuillam Thanks for the quick answer!
Unfortunately, the issue perstists with using `npm-groovy-lint@beta`
Output `npm-groovy-lint --version`:
```
GroovyLint: Successfully processed CodeNarc: 
CodeNarc version 3.5.0
npm-groovy-lint version 14.6.1-beta202408252305.0

Embeds:
CodeNarc version 3.5.0
- Groovy version 3.0.9 (superlite)
```

I played around a little bit and realised this issue do not come up if I run `npm-groovy-lint` with option `--no-parse`.
`npm-groovy-lint --no-parse`

Could you explain the impact of it?
If I disable it permanently, do I have any disadvantages in the linting? :-)
Thanks!

**Comment by nvuillam** [comment]
@Fabi9199 the Groovy files parsing uses Groovy 3.9.0 parser to confirm that the code is correct

What version of groovy are you using ?

Would it be possible to share an example of your Jenkinsfile so I can reproduce ?

**Comment by Fabi9199** [comment]
@nvuillam I currently do not know which Groovy Version our Jenkins is using. I will try to find out and come back to you.

I created a repository containing one of the files, which are failing with the mentioned exception:
https://github.com/Fabi9199/groovy-lint-test

Just run `npm-groovy-lint` :)

**Comment by nvuillam** [comment]
I succeeded to reproduce the issue, thanks !

@stevenh I don't find how to activate reports to see when it crashes during the parsing ... help ? ^^

@Fabi9199 meanwhile, good idea to disable with --no-parse :)

**Comment by Fabi9199** [comment]
Great! You`re welcome, I really appreciate your support here, thanks! :)

Do you have any hint, how I can enable the `--no-parse` option in the VSCode extension for npm-groovy-lint?

**Comment by nvuillam** [comment]
@Fabi9199 you can't for now but I'm currently making updates to VsCode Groovy Lint so I'll add this option !

**Comment by nvuillam** [comment]
@Fabi9199 I released 15.0.1 with @stevenh 's fix, please can you confirm it's ok now ? :)

Note: it's not propagated yet to VsCode Groovy Lint, only via the command line

**Comment by Fabi9199** [comment]
> @Fabi9199 I released 15.0.1 with @stevenh 's fix, please can you confirm it's ok now ? :)
> 
> Note: it's not propagated yet to VsCode Groovy Lint, only via the command line

@nvuillam I tried it just now and its working perfectly fine! Many thanks to both of you :)
Finally my action is working!

**Comment by nvuillam** [comment]
@Fabi9199 glad your use case is ok now :)

Thanks for reporting the issue :)
